tsl/src/compression/api.c

```diff
@@ -261,21 +261,13 @@ static void
 get_hypertable_or_cagg_name(Hypertable *ht, Name objname)
 {
 	ContinuousAggHypertableStatus status = ts_continuous_agg_hypertable_status(ht->fd.id);
-	if (status == HypertableIsNotContinuousAgg || status == HypertableIsRawTable)
-		namestrcpy(objname, NameStr(ht->fd.table_name));
-	else if (status == HypertableIsMaterialization)
+	if (status.isMaterialization)
 	{
 		ContinuousAgg *cagg = ts_continuous_agg_find_by_mat_hypertable_id(ht->fd.id);
 		namestrcpy(objname, NameStr(cagg->data.user_view_name));
 	}
 	else
-	{
```

Comment:
Is this state no longer possible?

Reply:
* the method is named `get_hypertable_or_cagg_name` 
* there is a passed in `Hypertable` typed argument which is not null for sure (line 263 dereferences it without thinking)
* ...so we always have a `Hypertable` which does have a name
* ...if I answer the cases in which the `_or_cagg` should be returned ; for the remaining cases the answer must be the`Hypertable`'s name